src/legendre_polynomial.cpp

```diff
@@ -0,0 +1,83 @@
+#include <tallies/legendre_polynomial.hpp>
+
+#include <iostream>
+
+LegendrePolynomials::LegendrePolynomials(std::size_t order)
+    : legendre_coeff_(), order_(order) {
+  // evaluate the coefficient for each and every order
+  // from zero to given order
+
+  // get the total number of coefficient to store
+  std::size_t coeff_length = 1;
+  if (order_ != 0) {
+    if (order_ % 2 == 0) {
+      const std::size_t length =
+          static_cast<std::size_t>((static_cast<double>(order_) - 1.) * 0.5 + 1.);
+      coeff_length =
+          length * (length + 1) + static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
+    } else {
+      const std::size_t length = static_cast<std::size_t>(static_cast<double>(order_) * 0.5 + 1);
+      coeff_length = length * (length + 1);
+    }
+  }
+  legendre_coeff_.reserve(coeff_length);
+
+  double inv_2 = 1.;
+  std::size_t low_pow = 1;
+
+  for (std::size_t i = 0; i <= order_; i++) {
+    if (low_pow == 1) {
+      low_pow = 0;
+    } else if (low_pow == 0) {
+      low_pow = 1;
+    }
+    std::size_t high_pow = static_cast<std::size_t>(static_cast<double>(i) * 0.5);
+    double sign_change = 1.;
+    if (high_pow % 2 == 1) sign_change = -1.;
+
+    for (std::size_t j = 0; j <= high_pow; j++) {
+      std::size_t k = high_pow - j;
+      double value = inv_2 * sign_change * coeff_factorial_evaluation(i, k);
+      sign_change *= -1;
+      legendre_coeff_.push_back(value);
+    }
+    inv_2 *= 0.5;
+  }
+
+  std::vector<double> sol = evaluate_legendres(0.1);
+  for (std::size_t i = 0; i < sol.size(); i++) {
+    std::cout << "order = " << i << "\t\t" << sol[i] << "\n";
```

Comment:
We should not have an std::cout here.

Reply:
I apologies here. It should not be here.